packages/excalidraw/fractionalIndex.ts

```diff
@@ -0,0 +1,264 @@
+  generateNJitteredKeysBetween as _generateNJitteredKeysBetween,
+  base62CharSet as _base62CharSet,
+  indexCharacterSet,
+} from "fractional-indexing-jittered";
+import { ENV } from "./constants";
+import { InvalidFractionalIndexError } from "./errors";
+
+type FractionalIndex = ExcalidrawElement["fractionalIndex"];
+
+export const base36CharSet = indexCharacterSet({
+  chars: "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ",
+  firstPositive: "A",
+  mostPositive: "Z",
+});
+
+const { chars, firstPositive, mostPositive, mostNegative } = _base62CharSet();
+
+const base62CharSet = indexCharacterSet({
+  chars,
+  firstPositive,
+  mostPositive,
+  mostNegative,
+});
+
+const charSet =
+  import.meta.env.DEV || import.meta.env.MODE === ENV.TEST
+    ? base36CharSet
+    : base62CharSet;
+
+export const generateKeyBetween = (
+  lower: string | null,
+  upper: string | null,
+) =>
+  import.meta.env.DEV || import.meta.env.MODE === ENV.TEST
+    ? _generateKeyBetween(lower, upper, charSet)
+    : _generateJitteredKeyBetween(lower, upper, charSet);
+
+export const generateNKeysBetween = (
+  lower: string | null,
+  upper: string | null,
+  n: number,
+) =>
+  import.meta.env.DEV || import.meta.env.MODE === ENV.TEST
+    ? _generateNKeysBetween(lower, upper, n, charSet)
+    : _generateNJitteredKeysBetween(lower, upper, n, charSet);
+
+/**
+ * Order the elements based on the fractional indices.
+ * - when fractional indices are identical (in prod only during jitter collisions), break the tie based on the element.id.
+ * - when there is no fractional index in one of the elements, respect the order of the array.
+ */
+export const orderByFractionalIndex = (elements: ExcalidrawElement[]) => {
+  return elements.sort((a, b) => {
+    if (a.fractionalIndex && b.fractionalIndex) {
+      if (a.fractionalIndex < b.fractionalIndex) {
+        return -1;
+      } else if (a.fractionalIndex > b.fractionalIndex) {
+        return 1;
+      }
+
+      // break ties based on the element id
+      return a.id < b.id ? -1 : 1;
+    }
+
+    // respect the order of the array
+    return 1;
+  });
+};
+
+/**
+ * Restore the fractional indices by mutating @param elements such that
+ * every element in the array has a fractional index smaller than its successor's.
+ *
+ * Neighboring indices (altough correct) might get updated as well.
+ *
+ * Only use this function when restoring or as a fallback to guarantee fractional
+ * indices consistency.
+ */
+export const restoreFractionalIndices = (
+  elements: readonly ExcalidrawElement[],
+) => {
+  for (const [index, element] of elements.entries()) {
+    const predecessorIndex = elements[index - 1]?.fractionalIndex || null;
+    const successorIndex = elements[index + 1]?.fractionalIndex || null;
+
+    if (
+      !isValidFractionalIndex(
+        element.fractionalIndex,
+        predecessorIndex,
+        successorIndex,
+      )
+    ) {
+      const fractionalIndex = restoreFractionalIndex(
```

Comment:
@ryan-di there were several try catch blocks in this file just console logging the caught error in case something went wrong. Such an approach can lead to silencing the errors on the client, without us being able to catch it in the monitoring logs. Therefore these try catch blocks were removed - in case there is an error thrown, it will be caught somewhere above, logged with the whole context & we will have a better chance of fixing it before anyone notices.